Sources/BraveWallet/Crypto/Portfolio/AddCustomAssetView.swift

```diff
@@ -318,21 +333,33 @@ struct AddCustomAssetView: View {
       if let tokenIdValue = Int16(tokenId) {
         tokenIdToHex = "0x\(String(format: "%02x", tokenIdValue))"
       }
-      token = BraveWallet.BlockchainToken(
-        contractAddress: addressInput,
-        name: nameInput,
-        logo: "",
-        isErc20: false,
-        isErc721: network.coin != .sol && !tokenIdToHex.isEmpty,
-        isNft: true,
-        symbol: symbolInput,
-        decimals: 0,
-        visible: true,
-        tokenId: tokenIdToHex,
-        coingeckoId: coingeckoId,
-        chainId: network.chainId,
-        coin: network.coin
-      )
+      if let knownERC721Token = tokenNeedsTokenId {
+        knownERC721Token.tokenId = tokenIdToHex
+        knownERC721Token.chainId = networkSelectionStore.networkSelectionInForm?.chainId ?? BraveWallet.MainnetChainId
```

Comment:
Not sure we want to fallback to ethereum mainnet here, I think we should use `knownERC721Token.chainId` as the fallback?